### DISCUSSION TITLE
Custom warm-up

### DISCUSSION THREAD

**Discussion by bppn** [Dotnet-Boxed/Templates#289] [Q&A]
﻿**Which project template is the issue about?**

Dotnet-Boxed GraphQL

**Which version of the project template is the issue about?**

current version

**Which version of Visual Studio or the dotnet SDK were you using?**

VS 2017 Professional

I'am deploying a GraphQL API on Azure App Service.

To provide a performant solution for the end-users, I want to warm-up the GraphQL service. Now the first call has a performance penalty.

I cannot figure out how this warm-up endpoint is able to load an initial query in the current boxed graphql solution.

All suggestions are welcome!

For documentation:

https://docs.microsoft.com/en-us/azure/app-service/deploy-staging-slots

<system.webServer>
    <applicationInitialization>
        <add initializationPage="/" hostName="[app hostname]" />
        <add initializationPage="/Home/About" hostName="[app hostname]" />
    </applicationInitialization>
</system.webServer>

**Comment by RehanSaeed** [comment]
Not entirely sure what you are asking.

The GraphQL editor makes an introspection query on load. You can figure out what the call is in Chrome DevTools, then use that as your warmup query.

Feel free to reopen if I have misunderstood.
